src/utils/request-queue.js

```diff
@@ -20,7 +20,7 @@ class RequestQueue {
       : false; // Defaults to 'false'
 
     // Mark if page environment is unloading
-    helpers.addEventListener('beforeunload', () => {
+    helpers.addEventListener('visibilitychange', () => {
```

Comment:
What do you think about checking the `visibilityState` here? I think we only want to do this when it's `hidden`

```
 if (document.visibilityState === 'hidden') {
   ...
}
```